### PR TITLE
Add Debug Panel for Simulated Data & HighlightedContent Test

### PR DESCRIPTION
**Pull request by CFIALeronB**
- Integrated a debug panel to toggle between live and simulated data.
- Added test for `highlightWords` in `HighlightedContent`, validating highlighted indices for the first data item.

#18
#30

### CODE REVIEW

File: src/api/useApiUtil.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // useApiUtil.tsx
 import { useState, useEffect } from "react";
+import testData from "../test_data/testData.json"; // Make sure the path points to your test data file
```
Comment (rngadam) [inline src/api/useApiUtil.tsx:3]: that only works at build time for a specific test data. I want us to have a bit more leeway on what data is returned. Why not make this a text field containing the JSON in the debug panel that we can set with some reasonable default samples?

**Review by rngadam — CHANGES_REQUESTED** [review]

**Review by rngadam — APPROVED** [review]
Our search results are minimum 10 results. Please open an issue to improve this test data and provide additional samples of query/example results. We'll use this to compare with the actual results to have something to shoot for.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -28,4 +28,3 @@ jobs:
       container-name: ${{ github.event.repository.name }}
       tag: ${{ github.sha }}
     secrets: inherit
-
```
Comment (rngadam) [inline .github/workflows/pull-request.yml:31]: https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

**Review by rngadam — CHANGES_REQUESTED** [review]

**Review by rngadam — APPROVED** [review]
LGTM. I'm assuming this is the version before you switched to fetching from finesse-data.

Would suggest you readthrough your code with @RussellJimmies in videoconference so he can make sure nachet-frontend follows some of the same practices.

**Review by rngadam — CHANGES_REQUESTED** [review]
oops, just saw that the tests are failing, they should be fixed before we merge

**Comment by rngadam** [comment]
@CFIALeronB tests still failing

**Comment by CFIALeronB** [comment]
@rngadam tests are now all passing, except the: lint-test / build Expected — Waiting for status to be reported. Not quite sure what needs to be done for that. This status is also present for the other pull requests: #32 , #27, #20

**Comment by rngadam** [comment]
> @rngadam tests are now all passing, except the: lint-test / build Expected — Waiting for status to be reported. Not quite sure what needs to be done for that. This status is also present for the other pull requests: #32 , #27, #20

@redat97

**Comment by redat97** [comment]
Fixed @CFIALeronB @rngadam I think I used the wrong parameter again not sure I fully understand how it works yet since it worked on the other PR I tried it on but looks like I need "lint-test / lint-test" as the source and not "lint-test / build" it's been manually fixed for now but I will document myself on that and fix the script again

**Review by rngadam — APPROVED** [review]

**Pull request merged**